docs/smart-contracts/guides/conventions/work-contractspec-js.mdx

```diff
@@ -4,4 +4,696 @@ hide_table_of_contents: true
+      cases: [
+        new xdr.ScSpecUdtErrorEnumCaseV0({
+          name: "InternalError",
+          doc: "Internal error occurred.",
+          value: 1,
+        }),
+        new xdr.ScSpecUdtErrorEnumCaseV0({
+          name: "UnauthorizedError",
+          doc: "Unauthorized access.",
+          value: 2,
+        }),
+        new xdr.ScSpecUdtErrorEnumCaseV0({
+          name: "InsufficientBalanceError",
+          doc: "Insufficient balance for the operation.",
+          value: 3,
+        }),
+        new xdr.ScSpecUdtErrorEnumCaseV0({
+          name: "OverflowError",
+          doc: "Overflow error occurred.",
+          value: 4,
+        }),
+      ],
+    }),
+  ),
+];
+
+// Initialize the contract specification
+const contractSpec = new Spec(contractSpecEntries);
+
+console.log("Contract Spec:", contractSpec);
+```
+
+**Explanation:**
+
+- We define the contract specification entries for each function and error in the `TokenManager` contract.
+- We create instances of `ScSpecFunctionV0` and `ScSpecFunctionInputV0` to define the functions and their inputs/outputs.
+- We create instances of `ScSpecUdtErrorEnumV0` and `ScSpecUdtErrorEnumCaseV0` to define the custom errors.
+- Finally, we initialize a `Spec` instance with the defined entries.
+
+Now you have the complete contract specification in JavaScript. You can use this specification to interact with your Soroban smart contract.
+
+With these steps, you have learned how to define a contract specification from scratch using JavaScript. You can proceed with deploying the contract, generating TypeScript bindings, and using the specification to interact with the contract.
+
+#### Option 2 Get the Spec from a Deployed Contract.
+
+##### Step 1: Deploy the Contract
+
+First, deploy your Soroban smart contract to the network.
+
+```bash
+soroban contract deploy --source orbit --network testnet --wasm token_manager.wasm
+```
+
+This command will return the contract ID, for example: `CBEUVPRZKMCX5MPUAF2JMKMRIFEXXJXQXBLUUW6LMN5GDDQQSX3UHYXA`.
+
+##### Step 2: Generate TypeScript Bindings
+
+Use the Soroban CLI to generate TypeScript bindings which use the contract's schema specifications to provide type-checked contract methods.
+
+```bash
+soroban contract bindings typescript --output-dir testbindings --network testnet --contract-id CBEUVPRZKMCX5MPUAF2JMKMRIFEXXJXQXBLUUW6LMN5GDDQQSX3UHYXA
+```
+
+This command will create TypeScript bindings in the specified directory.
+
+Alternatively, you can get your spec using javascript: credit to [@mason in Stellar Developers discord](https://discord.com/channels/897514728459468821/966788672164855829/1201708320336138271) and [his gist here] (https://gist.github.com/masonforest/27ebfafa3452227937b842c56731d817)
+
+Save this file as `readContractSpec.js`
+
+```typescript
+import fs from "node:fs/promises";
+import { Contract, xdr } from "@stellar/stellar-sdk";
+import jsXdr from "@stellar/js-xdr";
+
+const { XdrReader } = jsXdr;
```

Comment:
I think you can actually pull `XdrReader` from the SDK because it's exported by base!
```suggestion
import { Contract, xdr, XdrReader } from "@stellar/stellar-sdk";
```